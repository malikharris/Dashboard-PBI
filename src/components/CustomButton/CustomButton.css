/** Button Styles */

.buttonView {
  width: 229px;
  height: 50px;
  border-radius: 4px;
  background-color: #008248;
  margin-right: auto;
  margin-left: auto;
}

.buttonText {
  width: auto;
  height: 16px;
  margin: auto;
  color: white;
  font-family: Forza;
  font-size: 14px;
  font-weight: 500;
  /* text-transform: uppercase; */
}

.grid-btn {
  margin: 10px;
  padding: 3px;
  width: 100px;
  height: 30px;
  border-radius: 4px;
  border: 1px solid #bbbdc0;
  /* background-color: rgba(28, 155, 211, 0.1); */
}

/* .grid-btn:focus {
  outline: 1px solid var(--primary-color);
} */

.grid-btn-text {
  /* color: #1c9bd3; */
  font-family: Interstate;
  font-size: 13px;
  font-weight: 700;
  margin-right: 8px;
}
