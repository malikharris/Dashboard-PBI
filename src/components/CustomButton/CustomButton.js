import React from 'react';
import PropTypes from 'prop-types';

import columnsSvg from '../../Assets/images/columns.svg';
import filterSvg from '../../Assets/images/filter.svg';
import './CustomButton.css';

/**
 * @desc
 * isGridBtn: If true renders button with blue styling and an icon/img
 * img: An object with src and alt to render the grid btn img
 * icon: The icon imported from fontawesome if want to use svg icon instead of img
 * isFilter: Must be used in addition to isGridBtn. This adds default img/text
 * isColumns: Must be used in addition to isGridBtn. This adds default img/text
 * isRefresh: Must be used in addition to isGridBtn. This adds default icon/text
 * isFitScreen: Must be used in addition to isGridBtn. This adds default icon/text
 */
const CustomButton = props => {
  const {
    text,
    onClick,
    img,
    isGridBtn,
    isFilter,
    isColumns,
    icon,
    isRefresh,
    isFitScreen,
    isTextBtn,
    textColor,
  } = props;

  if (isGridBtn) {
    let gridBtnText = text;
    let gridBtnImg = img;
    let gridBtnIcon = icon;

    if (isFilter) {
      gridBtnText = 'Filter';
      gridBtnImg = { src: filterSvg, alt: 'Filter Icon' };
    } else if (isColumns) {
      gridBtnText = 'Columns';
      gridBtnImg = { src: columnsSvg, alt: 'Columns Icon' };
    } else if (isRefresh) {
      gridBtnText = 'Refresh';
    } else if (isFitScreen) {
      gridBtnText = 'Fit Screen';
    }

    return (
      <button
        type="button"
        className="grid-btn"
        style={{ backgroundColor: 'rgba(28, 155, 211, 0.1)' }}
        onClick={onClick}
      >
        <span className="grid-btn-text" style={{ color: '#1c9bd3' }}>
          {gridBtnText}
        </span>
        {gridBtnIcon ? (
          <div/>
        ) : (
          <img src={gridBtnImg.src} alt={gridBtnImg.alt} />
        )}
      </button>
    );
  }

  if (isTextBtn) {
    return (
      <button type="button" className="grid-btn" onClick={onClick}>
        <span className="grid-btn-text" style={{ color: textColor }}>
          {text}
        </span>
      </button>
    );
  }

  return (
    <button className="buttonView" type="button" onClick={onClick}>
      <p className="buttonText">{text}</p>
    </button>
  );
};

CustomButton.defaultProps = {
  text: 'Submit',
  img: null,
  isGridBtn: false,
  isFilter: false,
  isColumns: false,
  isRefresh: false,
  isFitScreen: false,
  icon: null,
  isTextBtn: false,
  textColor: 'black',
};

CustomButton.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  img: PropTypes.shape({
    src: PropTypes.number.isRequired,
    alt: PropTypes.string.isRequired,
  }),
  icon: PropTypes.any,
  isGridBtn: PropTypes.bool,
  isFilter: PropTypes.bool,
  isColumns: PropTypes.bool,
  isRefresh: PropTypes.bool,
  isFitScreen: PropTypes.bool,
  isTextBtn: PropTypes.bool,
  textColor: PropTypes.string,
};

export default CustomButton;
