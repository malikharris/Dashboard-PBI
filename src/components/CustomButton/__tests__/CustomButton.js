import React from 'react';
import { shallow } from 'enzyme';
import CustomButton from '../CustomButton';

let onClick;
let counter = 0;

beforeAll(() => {
  onClick = () => {
    counter++;
  };
});

it('should render correctly', () => {
  const wrapper = shallow(<CustomButton onClick={onClick} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render CustomButton correctly with default text', () => {
  const wrapper = shallow(<CustomButton onClick={onClick} />);
  expect(wrapper.find('p').props().children).toEqual('Submit');
});

it('should render CustomButton correctly with custom text', () => {
  const text = 'My Custom Button Title';
  const wrapper = shallow(<CustomButton onClick={onClick} text={text} />);
  expect(wrapper.find('p').props().children).toEqual(text);
});

it('should correctly use passed-in click handler', () => {
  counter = 0;
  const wrapper = shallow(<CustomButton onClick={onClick} />);
  wrapper.find('button').simulate('click');
  expect(counter).toEqual(1);
});
