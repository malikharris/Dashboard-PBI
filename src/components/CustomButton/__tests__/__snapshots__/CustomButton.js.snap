// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<button
  className="buttonView"
  onClick={[Function]}
  type="button"
>
  <p
    className="buttonText"
  >
    Submit
  </p>
</button>
`;
